Remotion/Web/Development.WebTesting/HostingStrategies/IisExpressHostingStrategy.cs

```diff
@@ -46,15 +44,19 @@ public IisExpressHostingStrategy ([NotNull] string webApplicationPath, int port)
     /// <param name="properties">The configuration properties.</param>
     [UsedImplicitly]
     public IisExpressHostingStrategy ([NotNull] NameValueCollection properties)
-        : this (
-            ArgumentUtility.CheckNotNull ("properties", properties)["path"],
-            int.Parse (ArgumentUtility.CheckNotNull ("properties", properties)["port"]))
+        : this (int.Parse (ArgumentUtility.CheckNotNull ("properties", properties)["port"]))
     {
     }
 
     /// <inheritdoc/>
-    public void DeployAndStartWebApplication ()
+    public void DeployAndStartWebApplication (ITestSiteConfiguration configuration)

```

Comment:
I see your points:

* It makes conceptually sense to pass site configs when trying to host the site, on the abstraction level of the interface
* It would apparently require some high grade technology magic to do it otherwise

Nonetheless, I can't asses if that really is what we want

Reply:
Okay, it would feel better to pass the testsite config into the ctor as fixed parameter #0. Since Port is also directly coupled with the hosting strategy, each instance of the hostring strategy is coupled to one specific test site (see option for stopping the test site in case cleanup is needed, and the inherent asymetry if we pass as argument) and finally the fact that the nullstrategy does not need to know about it.